spynnaker/pyNN/models/pynn_projection_common.py

```diff
@@ -267,6 +267,10 @@ def _get_synaptic_data(self, as_list, data_to_get):
         connection_holder.finish()
         return connection_holder
 
+    def _clear_cache(self):
+        post_vertex = self._projection_edge.post_vertex
+        post_vertex.clear_connection_cache()
```

Comment:
This could do with a check to ensure that the vertex actually has a method clear_connection_cache().  This could either be added to a new abstract class or else added to AbstractAcceptsIncomingSynapses.